src/python/pants/core/util_rules/partitions.py

```diff
@@ -73,10 +80,10 @@ class Partitions(FrozenDict["PartitionKeyT", "tuple[PartitionElementT, ...]"]):
 
     @classmethod
     def single_partition(
-        cls, elements: Iterable[PartitionElementT], key: PartitionKeyT = None  # type: ignore[assignment]
+        cls, elements: Iterable[PartitionElementT], key: PartitionKeyT | None = None
```

Comment:
(same about `None` I presume)

Reply:
This is also needed (as far as I can tell) - otherwise `mypy` complains about an incompatible default:
```
src/python/pants/core/util_rules/partitions.py:83:74: error: Incompatible default for argument "key" (default has type "None", argument has type "PartitionKeyT")  [assignment]
            cls, elements: Iterable[PartitionElementT], key: PartitionKeyT = None
```

Reply:
I guess it won't infer that `PartitionElementT` can be `None`? Lame.